pkg/grpc/server.go

```diff
@@ -120,8 +152,32 @@ func NewServersFromConfigurationAndServe(configurations []*configuration.ServerC
 			if err != nil {
 				return util.StatusWrapf(err, "Failed to create listening socket for %#v", listenPath)
 			}
-			go func() { serveErrors <- s.Serve(sock) }()
+			terminationGroup.Go(func() error {
+				if err := s.Serve(sock); err != nil {
+					return util.StatusWrap(err, "gRPC server failure")
+				}
+				return nil
+			})
 		}
 	}
-	return <-serveErrors
+	return nil
+}
+
+// SetReady marks the gRPC health service to respond with
+// grpc_health_v1.HealthCheckResponse_SERVING.
+func (sg *ServerGroup) SetReady() {
+	for i, h := range sg.healthServers {
+		h.SetServingStatus(sg.healthCheckServices[i], grpc_health_v1.HealthCheckResponse_SERVING)
+	}
+}
+
+// Shutdown stops the gRPC services. It cancels all active RPCs. Clients are
+// supposed to retry the operations whenever the service is up again.
+func (sg *ServerGroup) shutdown() {
```

Comment:
I think this part can now be removed, right? As in, we'll focus on making sure that storage gets flushed properly, but we won't actually tear down the RPC server.

Reply:
Kubernetes will reroute new connections to other pods during shutdown, but what will happen with open channels? They will continue as usual and then suddenly get "closed for writing" errors back, instead of "go away". Will Bazel retry on the same channel or make a new connection? I'm in favour of you suggestion as I don't see the point of a "perfect" shutdown, but just want to make sure we don't get into a bad state.